### ISSUE TITLE
Adding support for StatusTampered, battery level, and others

### ISSUE THREAD

**Issue by mattsheffiel** [hobbyquaker/homekit2mqtt#53]
I noticed there are comments in accessories/ContactSensor.js and others about adding support for StatusActive, StatusFault, and StatusTampered characteristics. Does this just mean additional if statements and appropriate code need to be added such as how statusLowBattery is implemented or does additional work need to be done to implement these?

Also, would it be possible at this point to add support for battery level (percentage) being reported? I played around with it but it seems that maybe support has to be added elsewhere in the code for homekit2mqtt before that can be added. This is my first time working with anything in NodeJS so I apologize in advance if these questions have obvious answers.

**Comment by hobbyquaker** [comment]
nothing to apologize for :) Yes, the Active/Fault/Tempered characteristics have to be added just to the accessories/*.js
About battery level: one could add extra characteristics to a service, but I fear the Home app will not show them. See the service definitions and the supported characteristics for each service here: https://github.com/KhaosT/HAP-NodeJS/blob/master/lib/gen/HomeKitTypes.js#L2509

**Comment by mattsheffiel** [comment]
Thanks for that link. Super useful!

After reading a bit more it seems that maybe I was using the wrong wording. It seems that there is a battery service with a battery level characteristic in addition to charging state, and low battery indicator. Would the Home app maybe display the percentage if that was used instead of adding a characteristic for battery level directly to the contact sensor service? It's not a huge priority for me just thought it would be useful down the road.

[https://github.com/KhaosT/HAP-NodeJS/blob/efe4c05e9b9ed831eeba89e4b67fddd8c70cc4ec/lib/gen/HomeKitTypes.js#L2587](url)

**Comment by hobbyquaker** [comment]
Ah got it. Yes, a Accessory can contain mutliple services, but I did a wrong design decision when making homekit2mqtt and until now every Accessory created by homekit2mqtt can only contain one service. Also I don't know how the Home App will present a multi-service accessory to the user, I guess each service will remain a single Widget in the app, but perhaps with some kind of linking between them? Afaik homebridge can create Accessories with multiple services, we could take a look how this is working there.
I created a new Issue so we don't forget about it, perhaps it would make sense to add this capability to homekit2mqtt, I'm not sure until now... https://github.com/hobbyquaker/homekit2mqtt/issues/56
